salt/utils/parsers.py

```diff
@@ -2739,31 +2733,31 @@ def _find_raet_minion(self, opts):
 
         role = opts.get('id')
         if not role:
-            emsg = ("Missing role required to setup RAET SaltCaller.")
-            logging.getLogger(__name__).error(emsg + "\n")
+            emsg = "Missing role required to setup RAET SaltCaller."
+            logger.error(emsg)
             raise ValueError(emsg)
 
         kind = opts.get('__role')  # application kind 'master', 'minion', etc
         if kind not in kinds.APPL_KINDS:
-            emsg = ("Invalid application kind = '{0}' for RAET SaltCaller.".format(kind))
-            logging.getLogger(__name__).error(emsg + "\n")
+            emsg = "Invalid application kind = '{0}' for RAET SaltCaller.".format(six.text_type(kind))
+            logger.error(emsg)
             raise ValueError(emsg)
 
-        if kind in [kinds.APPL_KIND_NAMES[kinds.applKinds.minion],
-                    kinds.APPL_KIND_NAMES[kinds.applKinds.caller], ]:
+        if kind in [kinds.APPL_KIND_NAMES[kinds.applKinds.minion], kinds.APPL_KIND_NAMES[kinds.applKinds.caller]]:
             lanename = "{0}_{1}".format(role, kind)
         else:
-            emsg = ("Unsupported application kind '{0}' for RAET SaltCaller.".format(kind))
-            logging.getLogger(__name__).error(emsg + '\n')
+            emsg = "Unsupported application kind '{0}' for RAET SaltCaller.".format(six.text_type(kind))
+            logger.error(emsg)
             raise ValueError(emsg)
 
         if kind == kinds.APPL_KIND_NAMES[kinds.applKinds.minion]:  # minion check
-            from raet.lane.yarding import Yard  # pylint: disable=3rd-party-module-not-gated
-            ha, dirpath = Yard.computeHa(dirpath, lanename, yardname)  # pylint: disable=invalid-name
-            if (os.path.exists(ha) and
-                    not os.path.isfile(ha) and
-                    not os.path.isdir(ha)):  # minion manor yard
-                return True
+            try:
+                from raet.lane.yarding import Yard  # pylint: disable=3rd-party-module-not-gated
+                ha, dirpath = Yard.computeHa(dirpath, lanename, yardname)  # pylint: disable=invalid-name
+                if os.path.exists(ha) and not os.path.isfile(ha) and not os.path.isdir(ha):  # minion manor yard
+                    return True
+            except ImportError as ex:
+                logger.error("Error while importing Yard: %s", six.text_type(ex))
```

Comment:
No need for `six.text_type(ex)`, this can just be `ex`. String replacement will automagically run `ex.__unicode__()` to get a string, because the string with the replacement placeholders is a unicode. For example:

```python
>>> u'The error is: %s' % TypeError(u'питон')
u'The error is: \u043f\u0438\u0442\u043e\u043d'
```